File: database/model/order.py
from sqlalchemy import Column,String, TIMESTAMP, Integer, VARCHAR, ForeignKey, Double
from sqlalchemy.orm import relationship
from database.model.base import Base
class Order(Base) :
    
    __tablename__ = 'order' # 테이블명
    __table_args__ = {'schema': 'luckydepot'} # 스키마명 

    id = Column(VARCHAR, primary_key=True) 
    user_seq = Column(Integer, ForeignKey('luckydepot.user.seq')) # 외래키 user seq
    payment_type = Column(String)
    price = Column(Double)
    address = Column(String)
    order_date = Column(TIMESTAMP)
    status = Column(String)
    delivery_type = Column(Integer)
    
    user = relationship("User", back_populates="order") # 관계 표시, sql쿼리 없이 관련된 테이블 접근 가능,  데이터 동기화

File: Lucky_Depot/Model/Order.swift
//
//  Order.swift
//  final_project
//
//  Created by Eunji Kim on 1/21/25.
//

struct Order: Decodable {
    var id: String
    var user_seq: Int
    var payment_type: String
    var price: Double
    var address: String
    var order_date: String
    var status: String
    
    init(id: String, user_seq: Int, payment_type: String, price: Double, address: String, order_date: String, status: String) {
        self.id = id
        self.user_seq = user_seq
        self.payment_type = payment_type
        self.price = price
        self.address = address
        self.order_date = order_date
        self.status = status
    }
}
